Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -827,12 +827,26 @@ void ContainerNode::removeChildren()
 
     Ref protectedThis { *this };
     NodeVector removedChildren;
-    removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren);
-
+#if PLATFORM(MAC)
+    if (this != &rootNode() && this->document().settings().opportunisticallyDeleteNodes() && this->document().settings().opportunisticSweepingAndGarbageCollectionEnabled()) {
```

Comment:
This will incur 4 indirect loads (tree-scope -> document -> settings -> bit flag).
We should consider cache this value in Document or TreeScope at the time of construction.